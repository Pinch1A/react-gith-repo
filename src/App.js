import { Route, Switch } from 'react-router'
import { BrowserRouter } from 'react-router-dom'
import { Home } from './components/Home/Home'
import PrivateRouter from './hoc/PrivateRouter'
import React from 'react'
import TokenValidator from './containers/TokenValidator/TokenValidation'
import { connect } from 'react-redux'
import styled from 'styled-components'

export const Button = styled.button`
  background-color: ${(props) => (props.primary ? 'orange' : 'grey')};
  color: white;
  border: none;
  border-radius: 5px;
  padding: 5px 10px;
  line-height: 31px;
  cursor: pointer;
`

const App = (props) => (
  <div className="App">
    <BrowserRouter>
      <Switch>
        <Route path="/token" component={TokenValidator} />
        <PrivateRouter path="/repo" token={props.token} />
        <Route path="/" component={Home} />
      </Switch>
    </BrowserRouter>
  </div>
)

const mapStateToProps = (state) => {
  return {
    token: state.tokenValidator.token,
  }
}

export default connect(mapStateToProps)(App)
