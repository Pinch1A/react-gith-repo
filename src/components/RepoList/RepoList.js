import { Button } from '../../App'
import React from 'react'
import { Repo } from '../index'
import styled from 'styled-components'

const Flex = styled.div`
  display: flex;
`
const ButtonWrapper = styled(Flex)`
  justify-content: center;
`
const StyledList = styled(Flex)`
  flex-direction: column;
  border: 1px solid #ccc;
  margin: 10px 0;
`

const RepoList = ({ pageInfo, fetchMoreRepo, repos }) => {
  return (
    <>
      <StyledList>
        {repos.map(({ node }, index) => (
          <Repo key={`${node.id}-${index}`} repository={node} />
        ))}
      </StyledList>
      <ButtonWrapper>
        {pageInfo && pageInfo.hasNextPage ? (
          <Button primary onClick={fetchMoreRepo}>
            Load More
          </Button>
        ) : (
          repos.length > 0 && (
            <Button href="#top">No more results,Go To Top</Button>
          )
        )}
      </ButtonWrapper>
    </>
  )
}

export default RepoList
