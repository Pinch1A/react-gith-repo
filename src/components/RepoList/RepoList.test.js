import { mount, shallow } from 'enzyme'
import { Button } from '../../App'
import React from 'react'
import RepoList from './RepoList'
const repos = [
  {
    node: {
      id: 'id-1',
      name: 'test-1',
      description: 'mock test 1',
      stargazers: { totalCount: 101 },
    },
  },
  {
    node: {
      id: 'id-2',
      name: 'test-2',
      description: 'mock test 2',
      stargazers: { totalCount: 102 },
    },
  },
]

describe('<RepoList>', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(
      <RepoList repos={repos} fetchMoreRepo={jest.fn()} pageInfo={{}} />
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('has more content to show', () => {
    const pageInfo = {
      hasNextPage: true,
    }
    const wrapper = shallow(
      <RepoList repos={repos} fetchMoreRepo={jest.fn()} pageInfo={pageInfo} />
    )
    const expected = 'Load More'
    expect(wrapper.find(Button).text()).toEqual(expected)
  })
  it('has not more content to show', () => {
    const pageInfo = {
      hasNextPage: false,
    }
    const wrapper = shallow(
      <RepoList repos={repos} fetchMoreRepo={jest.fn()} pageInfo={pageInfo} />
    )
    const expected = 'No more results,Go To Top'
    expect(wrapper.find(Button).text()).toEqual(expected)
  })
})
