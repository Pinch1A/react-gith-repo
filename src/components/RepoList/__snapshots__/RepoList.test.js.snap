// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoList> renders without crashing 1`] = `
<Fragment>
  <Styled(styled.div)>
    <Repo
      key="id-1-0"
      repository={
        Object {
          "description": "mock test 1",
          "id": "id-1",
          "name": "test-1",
          "stargazers": Object {
            "totalCount": 101,
          },
        }
      }
    />
    <Repo
      key="id-2-1"
      repository={
        Object {
          "description": "mock test 2",
          "id": "id-2",
          "name": "test-2",
          "stargazers": Object {
            "totalCount": 102,
          },
        }
      }
    />
  </Styled(styled.div)>
  <Styled(styled.div)>
    <styled.button
      href="#top"
    >
      No more results,Go To Top
    </styled.button>
  </Styled(styled.div)>
</Fragment>
`;
